module/VuFind/src/VuFind/Form/Form.php

```diff
@@ -759,14 +761,41 @@ public function getInputFilter()
         ];
 
         foreach ($this->getElements() as $el) {
-            $required = $el['type'] === 'checkbox'
-               ? $el['required'] ?? $el['requireOne'] ?? false
-               : $el['required'] ?? false;
+            $isCheckbox = $el['type'] === 'checkbox';
+            $required = $el['required']
+                ?? ($isCheckbox && ($el['requireOne'] ?? false));
+            $requireOne = $isCheckbox && ($el['requireOne'] ?? false);
```

Comment:
Rather than doing this calculation twice, what if you rearranged it to:

<pre>
$requireOne = $isCheckbox && ($el['requireOne'] ?? false);
$required = $el['required'] ?? $requireOne;
</pre>